Assignment_1.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+import sys
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import Normalizer
+
+
+def print_heading(title):
+    print("*" * 80)
+    print(title)
+    print("*" * 80)
+    return
+
+
+def main():
+    pd.set_option("display.max_rows", 10)
+    pd.set_option("display.max_columns", 500)
+    pd.set_option("display.width", 1_000)
+
+    col_names = [
+        "Sepal_Length",
+        "Sepal_Width",
+        "Petal_Length",
+        "Petal_Width",
+        "Species",
+    ]
+    # Load the Iris data set into Pandas DataFrame
+    iris_df = pd.read_csv("Iris.data", names=col_names)
+    print_heading("Printing first 10 rows")
+    print(iris_df.head(10))
+
+    # drop missing values
+    iris_df = iris_df.dropna()
+
+    iris_np = np.array(iris_df)
+
+    # summary statistics
+    print_heading("Summary Statistics using Numpy:")
+    print("Mean:", np.mean(iris_np[:, :-1], axis=0))
+
+    print("Max:", np.max(iris_np[:, :-1], axis=0))
+
+    print("Min:", np.min(iris_np[:, :-1], axis=0))
+
+    print("25%:", np.quantile(iris_np[:, :-1], 0.25, axis=0))
+
+    print("50%:", np.quantile(iris_np[:, :-1], 0.5, axis=0))
+
+    print_heading("Plots")
+    fig = px.scatter(
+        iris_df,
+        x="Sepal_Width",
+        y="Sepal_Length",
+        color="Species",
+        title="Scatter plot between Sepal_Width and Sepal_Length",
+    )
+    fig.show()
+
+    # Violin plot provides with density estimate on the y-axis
+    fig = px.violin(
+        iris_df,
+        y="Petal_Length",
+        x="Species",
+        color="Species",
+        box=True,
+        points="all",
+        hover_data=iris_df.columns,
+        title=" Violin plot using Petal_Length as feature",
+    )
+    fig.show()
+
+    # Boxplot using Petal_Length as feature to show what percentile
+    # ranges in what region
+    fig = px.box(
+        iris_df,
+        y="Petal_Length",
+        x="Species",
+        color="Species",
+        title=" Boxplot using Petal_Length as feature",
+    )
+    fig.show()
+
+    # Scatter Plot of different features
+    fig = px.scatter_matrix(
+        iris_df,
+        dimensions=["Sepal_Width", "Sepal_Length", "Petal_Width", "Petal_Length"],
+        color="Species",
+    )
+    fig.show()
+
+    # Pie chart to show Population of each Species in the dataset
+    fig = px.pie(
```

Comment:
Developing a pie chart here won't explain much, I would suggest trying something new like the 3d scatter plot. It is really good.

Reply:
The idea was to visualize the proportion of data for each Species.

Reply:
Okay.